var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -319,14 +319,16 @@ class Openmpi(AutotoolsPackage):
     depends_on('pmix', when='+pmix')
     depends_on('libevent', when='+pmix')
 
+    # PMIx is required for OpenMPI 4; omitting will use an internal copy
+    # Vendored version: depends_on('pmix@3.2.3', when='@4.1.2')
+    depends_on('pmix@3.2:', when='@4:')
```

Comment:
this will get more complicated soon as an eminent release of Open MPI uses features only available starting in PMIx 4.